index.js

```diff
@@ -47,34 +48,47 @@
     }
   }
 
-  // Use best RNG as possible
-  function randomBytes(n) {
-    var r;
-    if (typeof crypto !== 'undefined') {
-      if ((typeof buf === 'undefined') || ((bufIdx + n) > uuid.BUFFER_SIZE)) {
-        bufIdx = 0;
-        if (crypto.getRandomValues) {
-          buf = new Uint8Array(uuid.BUFFER_SIZE);
-          crypto.getRandomValues(bytes);
-        } else if (crypto.randomBytes) {
-          buf = crypto.randomBytes(uuid.BUFFER_SIZE);
-        } else {
-          throw new Error('Non-standard crypto library');
-        }
-      }
-      return buf.slice(bufIdx, bufIdx += n);
+  function randomBytesCryptoA(n) {
+    return crypto.randomBytes(n);
+  }
+
+  function randomBytesCryptoB(n) {
+    var bytes = new Uint8Array(n);
+    crypto.getRandomValues(bytes);
+    return bytes;
+  }
+
+  function randomBytesMath(n) {
+    var r = [];
+    for (i = 0; i < n; i++) {
+      r.push(getRandomInt(0, 255));
+    }
+    return r;
+  }
+
+  // Figure out which randomBytes function to use
+  if (typeof crypto !== 'undefined') {
+    if (crypto.randomBytes) {
+      randomBytes = randomBytesCryptoA;
+    } else if (crypto.getRandomValues) {
+      randomBytes = randomBytesCryptoB;
     } else {
```

Comment:
This else condition may need to be outside the `if (typeof crypto !== 'undefined')`. It should get executed even when `crypto === undefined`, I believe. Looks like currently it will not run if crypto is undefined.